main.cpp

```diff
@@ -176,7 +209,7 @@ DoubleType& DoubleType::multiply ( const double y)
 
 DoubleType& DoubleType::divide ( const double y) 
 { 
-    if (abs(y) > 0) 
+    if (std::abs(y) > 0) 
```

Comment:
Floating point division by zero is not an error.  Integer division by zero is an error.
- [ ] only warn.